lib/libshare/smb.c

```diff
@@ -138,10 +138,14 @@ smb_retrieve_shares(void)
 
 			if (strcmp(key, "path") == 0)
 				path = dup_value;
-			if (strcmp(key, "comment") == 0)
+			else if (strcmp(key, "comment") == 0)
 				comment = dup_value;
-			if (strcmp(key, "guest_ok") == 0)
+			else if (strcmp(key, "guest_ok") == 0)
 				guest_ok = dup_value;
+			else {
+				free(dup_value);
+				dup_value = NULL;
+			}
 
 			if (path == NULL || comment == NULL || guest_ok == NULL)
 				continue; /* Incomplete share definition */
```

Comment:
@behlendorf  @FransUrbo 
Increase a new function, 
Three reasons:
One: this is in the file of share_file_fp, there may be two "key" line, there will be a memory leak. 
Second: this is for safety, To free when parameter isn't NULL .
Three:  there are unified ZFS user mode code style.
Please review it again.